warp10/src/main/java/io/warp10/standalone/StandaloneIngressHandler.java

```diff
@@ -193,6 +196,8 @@ public StandaloneIngressHandler(KeyStore keystore, StandaloneDirectoryClient dir
     }
 
     this.maxValueSize = Long.parseLong(WarpConfig.getProperty(Configuration.STANDALONE_VALUE_MAXSIZE, DEFAULT_VALUE_MAXSIZE));
+    
+    this.isFDBStore = storeClient instanceof FDBStoreClient;
```

Comment:
This won't work for a sharded instance. You need to determine the backend from the configurationm checking the value of backend

Reply:
`this.isFDBStore = Constants.BACKEND_FDB.equals(WarpConfig.getProperty(Configuration.BACKEND))`

Reply:
a similar check should be added on line 285 of IngressStreamUpdateHandler (without the check for useFDB) and on line 322 of StandaloneStreamUpdateHandler (with the check that we use FDB).

Reply:
check added in both files